File: OneHandScheduling/solve/ruby/OneHandSchedulingSolver.rb
#! /usr/bin/env ruby

require_relative "OneHandScheduling"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  tStart = reader.next(TArray.new(TInt))
  reader.next()
  tEnd = reader.next(TArray.new(TInt))

  result = OneHandScheduling.new().checkSchedule(tStart, tEnd)
  IO.write(ARGV[1], Writer.new.next(result, TString).to_s)
end

init
main

File: OneHandScheduling/solve/python/OneHandSchedulingSolver.py
#! /usr/local/bin/python3

import os, sys
import OneHandScheduling

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        tStart = reader.next("int[]")
        reader.next()
        tEnd = reader.next("int[]")

    result = OneHandScheduling.checkSchedule(tStart, tEnd)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "String"))

if __name__ == "__main__":
    init()
    main()
